rpc/src/lib.rs

```diff
@@ -1,2 +1,192 @@
-mod rpc;
-pub use rpc::{fetch_prover_input, FetchProverInputRequest};
+use alloy::{
+    providers::Provider,
+    rpc::types::eth::{BlockId, Withdrawal},
+    transports::Transport,
+};
+use anyhow::Context as _;
+use evm_arithmetization::proof::{BlockHashes, BlockMetadata};
+use futures::{StreamExt as _, TryStreamExt as _};
+use itertools::{Either, Itertools as _};
+use prover::ProverInput;
+use serde::Deserialize;
+use serde_json::json;
+use trace_decoder::{
+    trace_protocol::{BlockTrace, BlockTraceTriePreImages, TxnInfo},
+    types::{BlockLevelData, OtherBlockData},
+};
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "snake_case")]
+#[allow(clippy::large_enum_variant)]
+enum JerigonTrace {
```

Comment:
We probably don't want this called `jerigon`. For context, the SMT stuff will live outside of the jerigon fork so in the nearish future this will be less obvious even internally, to add to the confusion the repo is still called erigon so anyone outside is going to be confused, even now.